### ISSUE TITLE
Issue with WAL after enabling Kerberos

### ISSUE THREAD

**Issue by Arttii** [confluentinc/kafka-connect-hdfs#37]
Hi,

So we had to enable Kerberos on the Cluster running CDH 5.5.
After adding the keytabs everything seems to work fine the files are being written. But if we say restart the connector agent we get this 

```
Caused by: java.io.IOException: Cannot obtain block length for LocatedBlock{BP-1540877157-10.240.188.16-1448467302331:blk_1074450611_710153; getBlockSize()=55; corrupt=false; offset=0; locs=[10.240.188.18:1004, 10.240.188.19:1004, 10.240.188.16:1004]; storageIDs=[DS-35b9ed3b-1614-43fd-b76b-af867857fafb, DS-d1b3c258-ec66-4b18-bffd-f14162aef547, DS-f1c65e45-6a79-4ef8-89e5-5cbf566b5468]; storageTypes=[DISK, DISK, DISK]}
    at org.apache.hadoop.hdfs.DFSInputStream.readBlockLength(DFSInputStream.java:359)
    at org.apache.hadoop.hdfs.DFSInputStream.fetchLocatedBlocksAndGetLastBlockLength(DFSInputStream.java:301)
    at org.apache.hadoop.hdfs.DFSInputStream.openInfo(DFSInputStream.java:238)
    at org.apache.hadoop.hdfs.DFSInputStream.<init>(DFSInputStream.java:231)
    at org.apache.hadoop.hdfs.DFSClient.open(DFSClient.java:1498)
    at org.apache.hadoop.hdfs.DistributedFileSystem$3.doCall(DistributedFileSystem.java:302)
    at org.apache.hadoop.hdfs.DistributedFileSystem$3.doCall(DistributedFileSystem.java:298)
    at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81)
    at org.apache.hadoop.hdfs.DistributedFileSystem.open(DistributedFileSystem.java:298)
    at io.confluent.connect.hdfs.wal.WALFile$Reader.openFile(WALFile.java:579)
    at io.confluent.connect.hdfs.wal.WALFile$Reader.<init>(WALFile.java:528)
    at io.confluent.connect.hdfs.wal.FSWAL.apply(FSWAL.java:107)
```

This seems to be relating to the connector WAL file in the logs directory. It like the file is not being closed and replicated properly, but I can access the file without problems. If i delete this and restart everything goes on. 

I only found smth related to this with 0 length files, but I cannot seem to find a proper solution because the log file does not seem to be zero length.

Has anyone experienced smth similar?

It seems like this is related

https://issues.apache.org/jira/browse/FLUME-2055

https://issues.apache.org/jira/browse/FLUME-2007

Basically I think we have to retry cosing the file if it failed like Flume does.

**Comment by Arttii** [comment]
Reinitiating the FS object solves this problem, but well thats seems like a huge hack.

**Comment by Ishiihara** [comment]
Thanks for reporting this. I will take a look and fix it in the connector.

**Comment by Ishiihara** [comment]
@Arttii I know this is filed quite a while ago but I would like to ask for some more clarification. What do you mean by reinitiating the FS object? Do you see this only when Kerberos is enabled or in simple authentication as well?

**Comment by cotedm** [comment]
@Arttii we have several users using the connector with Kerberos enabled and haven't heard of this issue resurfacing, so maybe it was fixed as a side effect to something else? If you still see it, please reopen and we can restart the conversation around what is happening in your environment.

**Comment by Arttii** [comment]
I somehow lost track of this. It was a weird bug connected to the Cloudera Hadoop jars we were using and the specific Java 1.8 version we were using. 

This was somehow connected to the HDFS caching problem sometimes happening with long running spark jobs as well. Turning of hdfs caching solved the issue as well.  But we ended up just changing the Hadoop and Java versions used
